mathics_scanner/tokeniser.py

```diff
@@ -470,12 +470,29 @@ def _skip_blank(self):
             if self.pos >= len(self.code):
                 if comment:
                     try:
+                        # Since we are inside a comment,
+                        # we can just do
+                        # ```
+                        # self.prescanner.incomplete()
+                        # self.code = self.prescanner.input_line
+                        # ```
+                        # avoiding the call to the method
+                        # `Prescanner.replace_escape_sequences()`
+                        # which have issues with escape characters...
                         self.incomplete()
                     except ValueError:
-                        # Funny symbols like | in comments can cause a ValueError.
-                        # Until we have a better fix -- like noting we are inside a
-                        # comment and should not try to substitute symbols -- ignore.
-                        pass
+                        # `incomplete` tries to parse substrings like `\|AAAAA`
+                        # that can be interpreted as a character reference.
+                        # To do that, it tries to get the
+                        # new line using the method
+                        # `Prescanner.replace_escape_sequences()`
+                        # Inside a comment, the special meaning of escape sequences
+                        # like `\|` should not be taken into account.
+                        #
+                        # In case of error, just let's pick the code
+                        # from the `input_line` attribute of
+                        # prescanner:
+                        self.code = self.prescanner.input_line
```

Comment:
Actually, we can replace the whole try/except block by lines 476 and 477.

Reply:
If that works, it would be simpler.

Reply:
I tested against the comment that produces the syntax error in `GS1.m` and works. Then I tested on each `GS?.m` and we still have one test wrong on `GS2.m` and `GS3.m`, but it seems it is not related to this issue.

Reply:
Looking again, at found a problem with the "simpler" approach: it could happen that after the comment finishes, some code that needs to take into account the escape characters comes. For example,
```
(*
Hi, this is a test \|<-this would produce an error if we pass by `replace_escape_sequences` 
*)
a="\|AAAAAA"
(*The previous line would require the use of replace_escape_sequences*)
```
So I was wrong with the extra simplification.

Reply:
Still this fix does not cover a corner case of the form
```
(* corner case *) A="\|AAAAAA"
```
So maybe we need to track this, and reload the last line. I will change the comment to mention this.

Reply:
> So maybe we need to track this, and reload the last line. I will change the comment to mention this.

Yes, please open an issue for this.